index.js

```diff
@@ -48,6 +48,53 @@ export class PProgress extends Promise {
 		});
 	}
 
+	static allSettled(promises, options) {
+		if (
+			options && typeof options.concurrency === 'number' &&
+			!(promises.every(promise => typeof promise === 'function'))
+		) {
+			throw new TypeError('When `options.concurrency` is set, the first argument must be an Array of Promise-returning functions');
```

Comment:
Promise-returning functions should always return promises no matter what.

Reply:
Asserting that doesn't seem to matter since the default behaviour of `Promise.allSettled` is also to assume that non-Promise values are resolved promises.

Reply:
I think you misunderstood. I'm saying that this should return a rejected promise instead of throwing.